opgee/processes/steam_generation.py

```diff
@@ -23,10 +23,28 @@
 
 
 class SteamGeneration(Process):
-
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        # TODO: this process disables itself under certain circumstances.
+        #  As noted in a comment below, this is undesirable behavior since
+        #  it silently edits the user's model. Rethink how to handle this.
+        #  In the meantime, it requires special processing below:
+        field = self.field
+        if field.steam_flooding == 1 and field.SOR != 0:
+            self._required_inputs = [
+                # TODO: avoid process names in contents.
+                "produced water for steam generation",
+                "makeup water for steam generation"
+            ]
+
+            self._required_outputs = [
+                "recycled water",
+            ]
+        else:
+            self._required_inputs = []
+            self._required_outputs = []
```

Comment:
There are potentially alternative ways to handle this, like subdividing the ``Process`` subclass so that each one has static stream definitions. This would introduce some redundancy across the variants, but that can be handled by refactoring the ``run`` method, breaking out common code in new methods. Either way, it's a bit more complicated.